spec/unit/provider/sensuctl_spec.rb

```diff
@@ -26,38 +26,38 @@
     end
 
     it 'should have JSON with type' do
-      subject.sensuctl_create('test', {name: 'test'}, {foo: 'bar'})
+      subject.sensuctl_create('Test', {name: 'test'}, {foo: 'bar'})
       j = JSON.parse(File.read(@tmp.path))
       expect(j['type']).to eq('Test')
     end
 
     it 'should have JSON with api_version' do
-      subject.sensuctl_create('test', {name: 'test'}, {foo: 'bar'})
+      subject.sensuctl_create('Test', {name: 'test'}, {foo: 'bar'})
       j = JSON.parse(File.read(@tmp.path))
       expect(j['api_version']).to eq('core/v2')
     end
 
     it 'should have JSON with metadata' do
-      subject.sensuctl_create('test', {name: 'test'}, {foo: 'bar'})
+      subject.sensuctl_create('Test', {name: 'test'}, {foo: 'bar'})
       j = JSON.parse(File.read(@tmp.path))
       expect(j['metadata']).to eq('name' => 'test')
     end
 
     it 'should have JSON with spec' do
-      subject.sensuctl_create('test', {name: 'test'}, {foo: 'bar'})
+      subject.sensuctl_create('Test', {name: 'test'}, {foo: 'bar'})
       j = JSON.parse(File.read(@tmp.path))
       expect(j['spec']).to eq({'foo' => 'bar'})
     end
 
     it 'should have JSON with only valid keys' do
-      subject.sensuctl_create('test', {name: 'test'}, {foo: 'bar'})
+      subject.sensuctl_create('Test', {name: 'test'}, {foo: 'bar'})
```

Comment:
Are all these capitalization changes for style or did something change with Sensu to make it necessary?

Reply:
Previously the function `sensuctl_create` was doing the capitalization but that breaks adding LDAP auth as `ldap` is the `type` and `Ldap` does not work.  Rather than add logic to disable the capitalization I just moved the logic to each provider so each provider passes the `type` that's required.

Reply:
The actual JSON passed to `sensuctl create -f` hasn't changed, just moved the logic to each provider rather than assuming the logic in shared function.